packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs

```diff
@@ -424,7 +425,7 @@ private ScmMethodProvider BuildProtocolMethod(MethodProvider createRequestMethod
 
             var methodSignature = new MethodSignature(
                 isAsync ? _cleanOperationName + "Async" : _cleanOperationName,
-                FormattableStringHelpers.FromString(Operation.Description),
+                DocHelpers.GetFormattableDescription(Operation.Summary, Operation.Doc) ?? FormattableStringHelpers.FromString(Operation.Name),
```

Comment:
Description of MethodSignature is nullable. Why do we mock the operation.Name as its description here?

Reply:
> Description of MethodSignature is nullable. Why do we mock the operation.Name as its description here?

It's because previously we use `Operation.Name` if `Description` is not available: https://github.com/microsoft/typespec/blob/b6dcb0482055a182b2545803b4c8d618fca6cfe8/packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/InputTypes/Serialization/TypeSpecInputOperationConverter.cs#L91

After discussion with Dapeng, we thought we should do less in the input type converters. So, I moved the logic in `TypeSepcInputOperationConverter` to the output model here.

We have similar changes elsewhere in this PR as well.